pkgs/by-name/ne/nexus-mods-app/package.nix

```diff
@@ -0,0 +1,81 @@
+{ buildDotnetModule
+, copyDesktopItems
+, desktop-file-utils
+, dotnetCorePackages
+, fetchFromGitHub
+, fontconfig
+, lib
+, libICE
+, libSM
+, libX11
+, nexus-mods-app
+, runCommand
+}:
+let
+  pname = "nexus-mods-app";
+  version = "0.4";
+  src = fetchFromGitHub {
+    owner = "Nexus-Mods";
+    repo = "NexusMods.App";
+    rev = "v${version}";
+    fetchSubmodules = true;
+    hash = "sha256-JKmRp7kSI7R0wkOvPVSChf8BNbMAPpnDHXr3h0plZGM=";
+  };
+in
+buildDotnetModule {
+  inherit pname src version;
+
+  projectFile = "src/NexusMods.App/NexusMods.App.csproj";
+
+  nativeBuildInputs = [
+    copyDesktopItems
+  ];
+
+  nugetDeps = ./deps.nix;
+
+  dotnet-sdk = dotnetCorePackages.sdk_8_0;
+  dotnet-runtime = dotnetCorePackages.runtime_8_0;
+
+  makeWrapperArgs = [ "--prefix PATH : ${lib.makeBinPath [desktop-file-utils]}" ];
```

Comment:
Adding `"--argv0"` to the wrapper args _might_ help fix the nxm handler desktop entry.

Reply:
I'm not entirely sure how to use it - adding `"--argv0 $out/bin/${nexus-mods-app.meta.mainProgram}"` to `makeWrapperArgs` didn't seem to fix `~/.local/share/applications/nexusmods-app-nxm.desktop`.

Reply:
The code checks at runtime for its executable, so that can not be fixed with exec -a (--argv0)

Reply:
I guess they're probably [doing something like this](https://stackoverflow.com/a/837501/2391965). If `argv[0]` isn't used, I guess we'll have to patch the source?

Maybe upstream would accept a PR allowing the exe location to be manually specified, e.g. via a cli option.